src/main/java/uk/gov/hmcts/ccd/data/roleassignment/DefaultRoleAssignmentRepository.java

```diff
@@ -0,0 +1,68 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.ccd.data.definition.DefaultCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ServiceException;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+@Qualifier(DefaultRoleAssignmentRepository.QUALIFIER)
+public class DefaultRoleAssignmentRepository implements RoleAssignmentRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultRoleAssignmentRepository.class);
+
+    public static final String QUALIFIER = "default";
+    public static final String AM_ERROR = "Problem getting Role Assignments from AccessManagement store because of %s";
+
+    private final ApplicationParams applicationParams;
+    private final SecurityUtils securityUtils;
+    private final RestTemplate restTemplate;
+
+    public DefaultRoleAssignmentRepository(final ApplicationParams applicationParams,
+                                           final SecurityUtils securityUtils,
+                                           @Qualifier("restTemplate") final RestTemplate restTemplate) {
+        this.applicationParams = applicationParams;
+        this.securityUtils = securityUtils;
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public RoleAssignmentResponse getRoleAssignments(String userId) {
+        try {
+            final HttpEntity<Object> requestEntity = new HttpEntity<>(securityUtils.authorizationHeaders());
+            final Map<String, String> queryParams = new HashMap<>();
+            queryParams.put("uid", ApplicationParams.encode(userId.toLowerCase()));
+
+            final String encodedUrl = UriComponentsBuilder.fromHttpUrl(applicationParams.amGetRoleAssignmentsURL())
+                .buildAndExpand(queryParams).toUriString();
+            return restTemplate.exchange(new URI(encodedUrl),
+                                         HttpMethod.GET, requestEntity,
+                                         RoleAssignmentResponse.class).getBody();
+
+        } catch (Exception e) {
+            LOG.warn("Error while retrieving Role Assignments", e);
+            if (e instanceof HttpClientErrorException
+                && ((HttpClientErrorException) e).getRawStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                throw new ResourceNotFoundException(String.format(AM_ERROR, e.getMessage()));
+            } else {
+                throw new ServiceException(String.format(AM_ERROR, e.getMessage()));
```

Comment:
ServiceException generates a 500 INTERNAL_SERVER_ERROR error code. What if the exception instead is of type 4XX BAD Request? We are converting it to a 500 which is misleading